test/integration/crcsuite/prepare.go

```diff
@@ -61,12 +61,19 @@ func DownloadBundle(bundleLocation string, bundleDestination string) (string, er
 func ParseFlags() {
 
 	flag.Parse()
-	if flag.NArg() < 2 {
+	if flag.NArg() < 3 {
 		fmt.Printf("Invalid number of arguments, the paths to the bundle file and to the pull secret file are required\n")
 		os.Exit(1)
 	}
-	bundleURL = flag.Args()[0]
-	_, bundleName = filepath.Split(bundleURL)
+
+	if flag.Args()[0] == "embedded" {
+		bundleEmbedded = true
+		bundleVersion = flag.Args()[2]
```

Comment:
From 'bundleVersion', don't we have a good idea of the expected bundleName? This would avoid the magic in `SetBundleName`

Reply:
@cfergeau We do, up to hypervisor - which is part of the name but not part of bundleVersion. If you happen to see a way that avoids `SetBundleName`, I'm all ears because I don't like it much either, but didn't manage to circumvent it. Thanks! I'll also have a think once more.

Reply:
If the bundle is not embedded, the full name was passed on the command line
If the bundle is embedded, we know the version, and I'd make the assumption that the hypervisor is the native one for the platform (ie not virtualbox). Similar to GetDefaultBundle().

Reply:
@cfergeau I changed it to what I think you meant. There are a few dodgy bits:

- The args to `make integration` are order sensitive. And I had to swap order since `BUNDLE_LOCATION` is optional now.
- I don't check for the situation when bundle is embedded and user still passes `BUNDLE_LOCATION`. As it is, if bundle is embedded, user must not specify bundle (that's how you tell tests that bundle is embedded). Not sure if that's good -- could be too implicit?
- bundle version is taken from Makefile, and not from the actual bundle. So the Makefile which comes with tests could contain a different version than the bundle, e.g. if user runs older CRC.

Some of these can be changed, e.g. the first one: I can probably pass bundle location to tests via tags. But this will be a separate PR as it affects other args too.